trieste/utils/pareto.py

```diff
@@ -101,10 +108,159 @@ def _bounds_2d(front: TensorType) -> BoundedVolumes:
+        pseudo_front = tf.concat([min_front, front, max_front], axis=0)
 
-        return BoundedVolumes(lower, upper)
+        pseudo_front_idx = tf.concat(
+            [
+                tf.zeros([1, number_of_objectives], dtype=tf.int32),
+                tf.argsort(front, axis=0) + 1,  # +1 as index zero is reserved for the ideal point
+                tf.ones([1, number_of_objectives], dtype=tf.int32) * len_front + 1,
+            ],
+            axis=0,
+        )
+
+        divide_conquer_cells = tf.stack(
+            [
+                tf.zeros(number_of_objectives, dtype=tf.int32),
+                (int(pseudo_front_idx.shape[0]) - 1)
+                * tf.ones(number_of_objectives, dtype=tf.int32),
+            ],
+            axis=0,
+        )[None]
+
+        total_size = tf.reduce_prod(max_front - min_front)
+
+        def while_body(
+            divide_conquer_cells: TensorType,
+            lower_result: TensorType,
+            upper_result: TensorType,
+        ):
+            divide_conquer_cells = tf.unstack(divide_conquer_cells, axis=0)
+            cell = divide_conquer_cells[-1]
+            divide_conquer_cells = tf.cond(
+                tf.not_equal(tf.size(divide_conquer_cells[:-1]), 0),
+                lambda: tf.stack(divide_conquer_cells[:-1]),
+                lambda: tf.zeros([0, 2, number_of_objectives], dtype=tf.int32),
+            )
+
+            arr = tf.range(number_of_objectives)
+            lower_idx = tf.gather_nd(pseudo_front_idx, tf.stack((cell[0], arr), -1))
+            upper_idx = tf.gather_nd(pseudo_front_idx, tf.stack((cell[1], arr), -1))
+            lower = tf.gather_nd(pseudo_front, tf.stack((lower_idx, arr), -1))
+            upper = tf.gather_nd(pseudo_front, tf.stack((upper_idx, arr), -1))
+
+            test_accepted = self._is_test_required((upper - jitter) < front)
+            lower_result, upper_result = tf.cond(
+                test_accepted,
+                lambda: self._accepted_test_body(lower_result, upper_result, lower_idx, upper_idx),
+                lambda: (lower_result, upper_result),
+            )
+
+            test_rejected = self._is_test_required((lower + jitter) < front)
+            divide_conquer_cells = tf.cond(
+                tf.logical_and(test_rejected, tf.logical_not(test_accepted)),
+                lambda: self._rejected_test_body(
+                    cell, lower, upper, divide_conquer_cells, total_size, threshold
+                ),
+                lambda: divide_conquer_cells,
+            )
+
+            return divide_conquer_cells, lower_result, upper_result
+
+        _, lower_result, upper_result = tf.while_loop(
+            lambda divide_conquer_cells, lower_result, upper_result: divide_conquer_cells.shape[0]
+            > 0,
+            lambda divide_conquer_cells, lower_result, upper_result: while_body(
+                divide_conquer_cells,
+                lower_result,
+                upper_result,
+            ),
```

Comment:
no need for the `lambda` here
```suggestion
            while_body,
```

Reply:
ah yeah missed that